docusaurus/docs/iOS/guides/push-notifications.md

```diff
@@ -2,135 +2,382 @@
-For testing purposes you can manage devices using Stream CLI:
+To receive push notifications from the Stream server the first step you need to do is register the device. To do this you need to call `UIApplication.shared.registerForRemoteNotifications()` and send the token from `application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data)`.
 
-### Adding devices
+Here is the boilerplate code that you can add to your `AppDelegate`:
 
-    stream chat:push:device:add
-
-### Removing devices
-
-    stream chat:push:device:delete
-
-### Getting the list of all devices registered for pushes
-    stream chat:push:device:get
+```swift
+func application(
+    _ application: UIApplication,
+    didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
+) {
+    guard ChatClient.shared.currentUserId != nil else {
+        log.warning("cannot add the device without connecting as user first, did you call connectUser")
+        return
+    }
 
-## Managing devices with StreamChat SDK
+    ChatClient.shared.currentUserController().addDevice(token: deviceToken) { error in
+        if let error = error {
+            log.warning("adding a device failed with an error \(error)")
+        }
+    }
+}
+```
 
-If you want for a user to receive push notifications, you need to request permissions for this. One of the most common places to do so is in `AppDelegate`'s `application(_:didFinishLaunchingWithOptions:)`, for instance like this:
+Because devices are linked to chat users, you should request the device token in the `connectUser` completion block
 
 ```swift
-func application(
-    _ application: UIApplication,
-    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
-) -> Bool {
+ChatClient.shared.connectUser(
+    userInfo: UserInfo(
+        id: "leia_organa",
+        name: "Leia Organa",
+        imageURL: URL(string: "https://cutt.ly/SmeFRfC")
+    ),
+    token: token
+) { error in
+    if let error = error {
+        log.error("connecting the user failed \(error)")
+        return
+    }
     UNUserNotificationCenter
-        .current() 
+        .current()
         .requestAuthorization(options: [.alert, .sound, .badge]) { granted, _ in
             if granted {
                 DispatchQueue.main.async {
                     UIApplication.shared.registerForRemoteNotifications()
                 }
             }
         }
-    return true
 }
 ```
-### Adding devices
 
-In case if the device was successfully registered for receiving push notifications, your app will receive an `application(_:didRegisterForRemoteNotificationsWithDeviceToken)` callback. At this point you should send an obtained device token to StreamChat's backend to finish the device's registration for push notifications.
+:::note
+Push notifications can be tricky to setup correctly, make sure to check for errors and settings on the [Dashboard](https://getstream.io/dashboard/)
```

Comment:
Add links to our content on Push Logging and 

- Push FAQ : https://getstream.io/chat/docs/ios-swift/push_-_common_issues_and_faq/?language=swift
- https://getstream.io/chat/docs/ios-swift/push_logs/?language=swift